apps/ogrlineref.cpp

```diff
@@ -1252,715 +1173,809 @@ static OGRErr GetCoordinates(OGRLayer *const poPkLayer, double dfPos,
-        else if (EQUAL(papszArgv[iArg], "-l"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            // coverity[tainted_data]
-            pszLineDataSource = papszArgv[++iArg];
-        }
-        else if (EQUAL(papszArgv[iArg], "-ln"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            // coverity[tainted_data]
-            pszLineLayerName = papszArgv[++iArg];
-        }
-        else if (EQUAL(papszArgv[iArg], "-lf"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
 #ifdef HAVE_GEOS
-            // coverity[tainted_data]
-            pszLineSepFieldName = papszArgv[++iArg];
-#else
-            fprintf(stderr,
-                    "GEOS support not enabled or incompatible version.\n");
-            exit(1);
+    argParser->add_argument("-create")
+        .flag()
+        .store_into(psOptions->bCreate)
+        .help(_("Create the linear reference file (linestring of parts)."));
 #endif
-        }
-        else if (EQUAL(papszArgv[iArg], "-p"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
-            // coverity[tainted_data]
-            pszPicketsDataSource = papszArgv[++iArg];
-        }
-        else if (EQUAL(papszArgv[iArg], "-pn"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
+
+    argParser->add_argument("-l")
+        .metavar("<src_line_datasource_name>")
+        .store_into(psOptions->osSrcLineDataSourceName)
+        .help(_("Name of the line path datasource."));
+
+    argParser->add_argument("-ln")
+        .metavar("<layer_name>")
+        .store_into(psOptions->osSrcLineLayerName)
+        .help(_("Layer name in the line path datasource."));
+
 #ifdef HAVE_GEOS
-            // coverity[tainted_data]
-            pszPicketsLayerName = papszArgv[++iArg];
-#else
-            fprintf(stderr,
-                    "GEOS support not enabled or incompatible version.\n");
-            exit(1);
+
+    argParser->add_argument("-lf")
+        .metavar("<field_name>")
+        .store_into(psOptions->osSrcLineSepFieldName)
+        .help(_("Field name for unique paths in layer."));
 #endif
-        }
-        else if (EQUAL(papszArgv[iArg], "-pm"))
-        {
-            CHECK_HAS_ENOUGH_ADDITIONAL_ARGS(1);
+
+    argParser->add_argument("-p")
+        .metavar("<src_repers_datasource_name>")
+        .store_into(psOptions->osSrcPicketsDataSourceName)
+        .help(_("Datasource of repers name."));
+
 #ifdef HAVE_GEOS
-            // coverity[tainted_data]
-            pszPicketsMField = papszArgv[++iArg];
-#else
-            fprintf(stderr,
-                    "GEOS support not enabled or incompatible version.\n");
-            exit(1);
+    argParser->add_argument("-pn")
+        .metavar("<layer_name>")
+        .store_into(psOptions->osSrcPicketsLayerName)
+        .help(_("Layer name in repers datasource."));
+
+    argParser->add_argument("-pm")
+        .metavar("<pos_field_name>")
+        .store_into(psOptions->osSrcPicketsSepFieldName)
```

Comment:
Shouldn't that be something like (osPicketsMField missing in the struct)
```suggestion
        .store_into(psOptions->osPicketsMField)
```

Reply:
It is weird that the test suite doesn't catch size. I see -pm is used in it, but perhaps the result isn't sufficiently checked

Reply:
I think that the test coverage might be improved.

Reply:
I have fixed the osPicketsMField issue.